File: python_project/sonarqube.py
import requests

# Define SonarQube URL and credentials
SONARQUBE_URL = 'http://localhost:9000/api/issues/search'
SONARQUBE_ADMIN_USERNAME = 'admin'
SONARQUBE_ADMIN_PASSWORD = 'Dhanush22@'
PROJECT_KEY = 'sonarintegration_python'

def fetch_sonar_results():
    try:
        # Make GET request to SonarQube API
        response = requests.get(SONARQUBE_URL, params={'componentKeys': PROJECT_KEY, 'resolved': "false"}, auth=(SONARQUBE_ADMIN_USERNAME, SONARQUBE_ADMIN_PASSWORD))
        response.raise_for_status()  # Raise exception for non-2xx status codes
        return response.json()  # Parse JSON response
    except requests.exceptions.RequestException as e:
        print(f'Error fetching SonarQube results: {e}')
        raise

# Example usage
if __name__ == '__main__':
    try:
        results = fetch_sonar_results()
        print(results)
    except Exception as e:
        print(f'Error: {e}')

File: sonarqube.js
// sonarqube.js
const axios = require('axios');
const SONARQUBE_URL = 'http://localhost:9000/api/issues/search';
const SONARQUBE_ADMIN_USERNAME = "admin";
const SONARQUBE_ADMIN_PASSWORD = 'Dhanush22@'
const PROJECT_KEY = 'sonarintegration_node';
async function fetchSonarResults() {



    try {

        const response = await axios.get(SONARQUBE_URL, {
            params: {
                componentKeys: PROJECT_KEY,
                resolved: false
            },
            auth: {
                username: SONARQUBE_ADMIN_USERNAME,
                password: SONARQUBE_ADMIN_PASSWORD
            }
        });
        // console.log(response.data)
        return response.data;
    } catch (error) {
        console.log(`Error fetching SonarQube results: ${error}`)
        throw new Error(`Error fetching SonarQube results: ${error}`);
    }
}



module.exports = { fetchSonarResults };
